tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/RabbitMQ/RabbitMQIntegration.cs

```diff
@@ -68,6 +67,45 @@ internal static Scope CreateScope(Tracer tracer, out RabbitMQTags tags, string c
             return scope;
         }
 
+        internal static void SetDataStreamsCheckpoint(Tracer tracer, Span span, RabbitMQTags tags, IBasicProperties basicProperties)
```

Comment:
Another comment on this method, could we split up this method into two, so one method is responsible for injecting context for producers and the other is responsible for extracting context for consumers? I know there's a little bit of overlap of the logic, but I think it might make the code easier to step through later if we more clearly separated the logic.